canopy-cover-enhancement/terra_ganEnhancement.py

```diff
@@ -0,0 +1,401 @@
+            elif fname.endswith('_left.bin'):
+                img_left = fname
+            elif fname.endswith('_right.bin'):
+                img_right = fname
+        if None in [img_left, img_right, terra_md_full]:
+            raise ValueError("could not locate all files & metadata in processing")
+
+        timestamp = resource['dataset_info']['name'].split(" - ")[1]
+
+        # Fetch experiment name from terra metadata
+        season_name, experiment_name, updated_experiment = get_season_and_experiment(timestamp, terra_md_full)
+        if None in [season_name, experiment_name]:
+            raise ValueError("season and experiment could not be determined")
+
+        # Determine output directory
+        self.log_info(resource, "Hierarchy: %s / %s / %s / %s / %s / %s / %s" % (
+        season_name, experiment_name, self.sensors.get_display_name(),
+        timestamp[:4], timestamp[5:7], timestamp[8:10], timestamp))
+        target_dsid = build_dataset_hierarchy_crawl(host, secret_key, self.clowder_user, self.clowder_pass,
+                                                    self.clowderspace,
+                                                    season_name, experiment_name, self.sensors.get_display_name(),
+                                                    timestamp[:4], timestamp[5:7], timestamp[8:10],
+                                                    leaf_ds_name=self.sensors.get_display_name() + ' - ' + timestamp)
+        left_gan_tiff = self.sensors.create_sensor_path(timestamp, opts=['left_gan'])
+        right_gan_tiff = self.sensors.create_sensor_path(timestamp, opts=['right_gan'])
+        left_mask_tiff = self.sensors.create_sensor_path(timestamp, opts=['left_mask'])
+        right_mask_tiff = self.sensors.create_sensor_path(timestamp, opts=['right_mask'])
+        uploaded_file_ids = []
+
+        # Attach LemnaTec source metadata to Level_1 product
+        self.log_info(resource, "uploading LemnaTec metadata to ds [%s]" % target_dsid)
+        remove_metadata(connector, host, secret_key, target_dsid, self.extractor_info['name'])
+        terra_md_trim = get_terraref_metadata(all_dsmd)
+        if updated_experiment is not None:
+            terra_md_trim['experiment_metadata'] = updated_experiment
+        terra_md_trim['raw_data_source'] = host + ("" if host.endswith("/") else "/") + "datasets/" + resource['id']
+        level1_md = build_metadata(host, self.extractor_info, target_dsid, terra_md_trim, 'dataset')
+        upload_metadata(connector, host, secret_key, target_dsid, level1_md)
+
+        #current_ratio, current_binMask, current_rgbMask = gen_cc_enhanced(fname)
+        gps_bounds = geojson_to_tuples(terra_md_full['spatial_metadata']['stereoTop']['bounding_box'])
+
+        if not file_exists(left_gan_tiff) or self.overwrite:
+            self.log_info(resource, "creating & uploading %s" % left_gan_tiff)
+            left_ratio, left_mask, left_gan = gen_cc_enhanced(img_left)
```

Comment:
so for calling gen_cc_enhanced(), outputs are ratio, binMask, rgbMask. so the 3rd output is what we want to preserve. you call it left_gan here but really it's left_rgb_mask.